network/openziti/pkg.yaml

```diff
@@ -0,0 +1,65 @@
+name: openziti
+variant: alpine
+shell: /toolchain/bin/bash
+install:
+  - unzip
+  - zip
+  - ninja-build
+  - ninja
+  - zlib-static
+dependencies:
+  - image: "ghcr.io/siderolabs/tools:v1.7.0-4-gc844dc3"
+steps:
+  - sources:
+      - url: https://github.com/openziti/ziti-tunnel-sdk-c/archive/refs/tags/{{ .OPENZITI_TUNNELER_VERSION }}.tar.gz
+        destination: edge-tunnel-source.tar.gz
+        sha256: d455672bf3b6ff28fd0ff864f868f7d6e3de99e6c666a120227fa9dab2d91f78
+        sha512: 5aeb05347381124e9d75693f12e0d234499a8c6482d322019576a24c622d986d43f159454a43c844623f171a456062a5e1afffcee7f7cbb03740216dbd3bb74c
+    env:
+      VCPKG_ROOT: /vcpkg
+      VCPKG_FORCE_SYSTEM_BINARIES: 1
+      CC: /toolchain/bin/gcc
+      CXX: /toolchain/bin/g++
+    prepare:
+      - |
+        # vcpkg depends on git as a delivery tool
+        # if to download .tar.gz from releases it would fail saying it cannot retrieve versions/baseline.json version using git show
+        # tried my best to find any CMAKE variable within vcpkg which disables this behaviour
+        # we fetch here commit to which tag 2024.07.12 is assigned
+        git clone https://github.com/microsoft/vcpkg.git /vcpkg
+        git --git-dir=/vcpkg/.git --work-tree=/vcpkg checkout 1de2026f28ead93ff1773e6e680387643e914ea1
+
+      - |
+        sed -i 's#$VERSION#{{ .VERSION }}#' /pkg/manifest.yaml
+
+      - |
+        mkdir -p /vcpkg-git /vcpkg /ziti-tunnel-sdk-c/build
```

Comment:
why are we building an sdk? doesn't make sense, shouldn't it be using this https://github.com/openziti/ziti/?

Reply:
> why are we building an sdk? doesn't make sense, shouldn't it be using this https://github.com/openziti/ziti/?

No we shouldn't. OpenZiti tunneler is exactly for hosting or tunneling. Main ziti repo is for the other part of the OpenZiti project which provides a controller, router and etc.

Reply:
Tunneler is just a daemon that runs on the host machines and allows to put the network communication on or from the overlay network it creates. For Talos its impossible to change routing from the extension, so the tunneler runs in a 'run-host' mode which allows offloading of a overlay traffic on the underlay network.